File: scripts/topic_subscriber_template.py
#! /usr/bin/env python
#coding:utf-8

import rospy
from std_msgs.msg import String

class TopicSubscriber():
    def __init__(self):
        self.sub_msg = rospy.Subscriber('pub_msg', String, self.callback_message )

    def callback_message(self, msg):
        rospy.loginfo('\n [ ' + rospy.get_name() + ' ] Message Subscribed\n%s\n', msg )
        return

if __name__ == '__main__':
    try:
        rospy.init_node('topic_subscriber_template')
        
        node = TopicSubscriber()

        rospy.spin()

    except rospy.ROSInterruptException: pass

File: src/topic_subscriber_template.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>
#include <iostream>

namespace ros_template_programs {
    class TopicSubscriber {
        private:
            ros::NodeHandle nh_;
            ros::NodeHandle pnh_;
            ros::Subscriber sub_msg_;
            void callbackMessage( const std_msgs::StringConstPtr &msg );

        public:
            TopicSubscriber( );
            void loopPublish();
    };
}

void ros_template_programs::TopicSubscriber::callbackMessage( const std_msgs::StringConstPtr &msg ) {
    ROS_INFO ( "\n[%s] Message Subscribed\n%s\n", ros::this_node::getName().c_str(), msg->data.c_str() );
    return;
}

ros_template_programs::TopicSubscriber::TopicSubscriber( ) : nh_(), pnh_("~") {
    sub_msg_ = nh_.subscribe( "pub_msg", 1, &ros_template_programs::TopicSubscriber::callbackMessage, this );
}

int main(int argc, char *argv[])  {
    ros::init(argc, argv, "topic_subscriber_template");
    ros_template_programs::TopicSubscriber topic_sub;
    ros::spin();
    return 0;
}